app/models/protocol.rb

```diff
@@ -294,6 +294,10 @@ def self.clone_contents(src, dest, current_user, clone_keywords)
       step.assets.each do |asset|
         asset2 = asset.dup
         asset2.save!
+        asset2.file.attach(io: StringIO.new(asset.file.download),
```

Comment:
I would use temp files instead of loading file content to memory as file size is limited by FILE_MAX_SIZE_MB and it can be quite big